partner_contact_depth/models/res_partner.py

```diff
@@ -0,0 +1,38 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.depends("parent_id")
+    def _compute_contact_depth(self):
+        for partner in self:
+            depth = 0
+            parent = partner.parent_id
+            while parent:
+                depth += 1
+                parent = parent.parent_id
+            partner.depth = depth
+
+    depth = fields.Integer(compute="_compute_contact_depth", store=True)
```

Comment:
I believe this may cause inconsistencies in the long run for specific scenarios. Notice that the `depth` of a partner will only be updated if the `parent_id` changes.

Then, if we have a hierarchy relation such as:
- Contact **A** with no parent.
- Contact **B**, being **A** its parent.
- Contact **C**, being **B** its parent.

If for some reason, the Contact **A** gets assigned a Parent now, the hierarchy will change and the `depth` will only be changed for Contact **A**, when it should be done for **B** and **C** as well.

Therefore, I'd say it should perhaps be a non-computed field which would get updated every time the parent_id changes for itself, and for its childs. Or, perhaps adding something like:

```suggestion
    @api.depends("parent_id")
    def _compute_contact_depth(self):
        for partner in self:
            depth = 0
            parent = partner.parent_id
            while parent:
                depth += 1
                parent = parent.parent_id
            partner.depth = depth
            partner.child_ids._compute_contact_depth()

    depth = fields.Integer(compute="_compute_contact_depth", store=True)
```

I guess this would work fine, but of course I would add some tests to reinforce the desired functionality.

Reply:
I refactored the depth computation in order to handle with reparenting. Thanks for your example :)